src/mysql-util/src/decoding.rs

```diff
@@ -174,6 +174,35 @@ fn val_to_datum<'a>(
                         temp_strs.push(val.to_string());
                         Datum::from(temp_strs.last().unwrap().as_str())
                     }
+                    Some(MySqlColumnMeta::Date) => {
+                        // Some MySQL dates are invalid in chrono/NaiveDate (e.g. 0000-00-00), so
+                        // we need to handle them directly as strings
+                        if let Value::Date(y, m, d, 0, 0, 0, 0) = value {
+                            temp_strs.push(format!("{:04}-{:02}-{:02}", y, m, d));
+                        } else {
+                            Err(anyhow::anyhow!(
+                                "received unexpected value for date type: {:?}",
+                                value
+                            ))?
+                        }
+                        Datum::from(temp_strs.last().unwrap().as_str())
+                    }
+                    Some(MySqlColumnMeta::Timestamp) => {
+                        // Some MySQL dates are invalid in chrono/NaiveDate (e.g. 0000-00-00), so
+                        // we need to handle them directly as strings
+                        if let Value::Date(y, m, d, h, mm, s, ms) = value {
+                            temp_strs.push(format!(
+                                "{:04}-{:02}-{:02} {:02}:{:02}:{:02}.{:06}",
+                                y, m, d, h, mm, s, ms
+                            ));
```

Comment:
Do we have the ability to get the timestamp precision here? It looks like the textual representation depends on the declared precision:

```
mysql> select cast('2024-01-01 00:00:00.1234' as datetime (4));
+--------------------------------------------------+
| cast('2024-01-01 00:00:00.1234' as datetime (4)) |
+--------------------------------------------------+
| 2024-01-01 00:00:00.1234                         |
+--------------------------------------------------+
1 row in set (0.00 sec)

mysql> select cast('2024-01-01 00:00:00.1234' as datetime (6));
+--------------------------------------------------+
| cast('2024-01-01 00:00:00.1234' as datetime (6)) |
+--------------------------------------------------+
| 2024-01-01 00:00:00.123400                       |
+--------------------------------------------------+
1 row in set (0.00 sec)

mysql> select cast('2024-01-01 00:00:00.1234' as datetime);
+----------------------------------------------+
| cast('2024-01-01 00:00:00.1234' as datetime) |
+----------------------------------------------+
| 2024-01-01 00:00:00                          |
+----------------------------------------------+
1 row in set (0.00 sec)
```

Reply:
Yes I can capture the column precision when we create the schema. Pushing a change to format based on the precision now